src/main/lib/osu-file-parser/OsuParser.ts

```diff
@@ -1,125 +1,345 @@
-import { AudioSource, ImageSource, ResourceID, Result, Song } from '../../../@types';
-import readline from 'readline';
-import path from 'path';
-import { fail, ok } from '../rust-like-utils-backend/Result';
-import { OsuFile } from './OsuFile';
-import { access, getFiles, getSubDirs } from '../fs-promises';
-import fs from 'graceful-fs';
-import { assertNever } from '../tungsten/assertNever';
-
-
+import fs from "graceful-fs";
+import os from "os";
+import readline from "readline";
+import { AudioSource, ImageSource, ResourceID, Result, Song } from "../../../@types";
+import { access } from "../fs-promises";
+import { fail, ok } from "../rust-like-utils-backend/Result";
+import { assertNever } from "../tungsten/assertNever";
+import { OsuFile } from "./OsuFile";
 
 const bgFileNameRegex = /.*"(?<!Video.*)(.*)".*/;
 const beatmapSetIDRegex = /([0-9]+) .*/;
 
-
-
-type FileState = 'Initial'
-  | 'NextState'
-  | 'General'
-  | 'Editor'
-  | 'Metadata'
-  | 'Difficulty'
-  | 'Events'
-  | 'TimingPoints'
-  | 'Colours'
-  | 'HitObjects';
+type FileState =
+  | "Initial"
+  | "NextState"
+  | "General"
+  | "Editor"
+  | "Metadata"
+  | "Difficulty"
+  | "Events"
+  | "TimingPoints"
+  | "Colours"
+  | "HitObjects";
 
 const OFFSET = 0;
 const BPM = 1;
 
-
-
 type Table<T> = Map<ResourceID, T>;
 
-export type DirParseResult = Promise<Result<[Table<Song>, Table<AudioSource>, Table<ImageSource>], string>>
+export type DirParseResult = Promise<
+  Result<[Table<Song>, Table<AudioSource>, Table<ImageSource>], string>
+>;
+
+// Overriding Buffer prototype because I'm lazy.
+// Should probably get moved to another file, or make a wrapper instead.
+
+Buffer.prototype.read_bytes = function (n) {
```

Comment:
Add a short description to each function. Example for `read_u8`:

Reads an unsigned byte (8-bit)